src/Redis/Orleans.GrainDirectory.Redis/Options/RedisGrainDirectoryOptions.cs

```diff
@@ -11,37 +12,51 @@ namespace Orleans.Configuration
     public class RedisGrainDirectoryOptions 
     {
         /// <summary>
-        /// Configure the Redis client
+        /// Gets or sets the Redis client configuration.
         /// </summary>
         [RedactRedisConfigurationOptions]
         public ConfigurationOptions ConfigurationOptions { get; set; }
 
+        /// <summary>
+        /// The delegate used to create a Redis connection multiplexer.
+        /// </summary>
+        public Func<RedisGrainDirectoryOptions, Task<IConnectionMultiplexer>> CreateMultiplexer { get; set; } = DefaultCreateMultiplexer;
+
         /// <summary>
         /// Entry expiry, null by default. A value should be set ONLY for ephemeral environments (like in tests).
         /// Setting a value different from null will cause duplicate activations in the cluster.
         /// </summary>
         public TimeSpan? EntryExpiry { get; set; } = null;
+
+        /// <summary>
+        /// The default multiplexer creation delegate.
+        /// </summary>
+        public static async Task<IConnectionMultiplexer> DefaultCreateMultiplexer(RedisGrainDirectoryOptions options) => await ConnectionMultiplexer.ConnectAsync(options.ConfigurationOptions);
```

Comment:
can probably nuke the `async` and `await` here; downside: won't appear in stack-trace if it 'splodes